lib/validator/validator.go

```diff
@@ -326,8 +329,26 @@ func validateIPPool(v *validator.Validate, structLevel *validator.StructLevel) {
 				}
 			}
 		}
+
+		// IP Pool CIDR cannot overlap with IPv4 or IPv6 link local address range.
+		_, ipv4LinkLocalNet, _ := net.ParseCIDR("169.254.0.0/16")
+		_, ipv6LinkLocalNet, _ := net.ParseCIDR("fe80::/10")
+
+		if pool.Metadata.CIDR.Version() == 4 && isNetOverlap(&pool.Metadata.CIDR.IPNet, ipv4LinkLocalNet) {
+			structLevel.ReportError(reflect.ValueOf(pool.Metadata.CIDR),
+				"CIDR", "", reason(overlapsV4LinkLocal))
+		}
+
+		if pool.Metadata.CIDR.Version() == 6 && isNetOverlap(&pool.Metadata.CIDR.IPNet, ipv6LinkLocalNet) {
+			structLevel.ReportError(reflect.ValueOf(pool.Metadata.CIDR),
+				"CIDR", "", reason(overlapsV6LinkLocal))
+		}
 	}
+}
 
+// isNetOverlap is a utility function that returns true if the two subnet have an overlap.
+func isNetOverlap(n1, n2 *net.IPNet) bool {
```

Comment:
don't really like the whole idea of layering on top of `net.IPNet` but I guess since we have it, I can do it for this one. In future, if we move away from this layering on top of `net`, this will come back here :)